.gitlab/package_build/build_agent_dmg.sh

```
@@ -124,19 +124,25 @@ if [ "$SIGN" = true ]; then
 
     # Apply timeout / retry
     for attempt in $(seq 1 $NOTARIZATION_ATTEMPTS); do
-    RESULT=$(timeout "$NOTARIZATION_TIMEOUT" xcrun notarytool submit --apple-id "$APPLE_ACCOUNT" --team-id "$TEAM_ID" --password "$NOTARIZATION_PWD" "$LATEST_DMG" --wait) || EXIT_CODE=$?
-    echo "Results: $RESULT"
-    SUBMISSION_ID=$(echo "$RESULT" | awk '$1 == "id:"{print $2; exit}')
-    echo "Submission ID: $SUBMISSION_ID"
-    echo "Submission logs:"
-    xcrun notarytool log --apple-id "$APPLE_ACCOUNT" --team-id "$TEAM_ID" --password "$NOTARIZATION_PWD" "$SUBMISSION_ID"
-    if [ -n "$EXIT_CODE" ]; then
-        echo "Notarization attempt #$attempt/$NOTARIZATION_ATTEMPTS failed, retrying in $NOTARIZATION_WAIT_TIME"
-        sleep "$NOTARIZATION_WAIT_TIME"
-    else
-        echo "Successfully notarized the package"
-        break
-    fi
+        EXIT_CODE=
+        RESULT=$(timeout "$NOTARIZATION_TIMEOUT" xcrun notarytool submit --apple-id "$APPLE_ACCOUNT" --team-id "$TEAM_ID" --password "$NOTARIZATION_PWD" "$LATEST_DMG" --wait) || EXIT_CODE=$?
```

If I understand correctly the EXIT_CODE here will be overwritten by the exit code of `xcrun notarytool log` below.

- Is that fine because we expect the `log` command to fail if `submit` failed? In that case I guess we could just ignore the error rather than setting EXIT_CODE.
- Should we instead consider the attempt failed at this stage and wrap it in error handling code of its own? In that case it might make sense to split both steps (each having its own set of retries, perhaps using the code in https://github.com/DataDog/datadog-agent/blob/main/tools/ci/retry.sh).

I think the current behavior might be wrong and that the second option might actually be the only correct way to do this. The reason being that if the submission succeeds and `log` fails, it might not make sense to repeatedly `submit` again, but to just retry the `log` call.

Looking at the docs, I also wonder why we use the `timeout` command instead of passing the `--timeout` flag that this command [seems to have](https://keith.github.io/xcode-man-pages/notarytool.1.html#timeout).